python/lsst/ts/scriptqueue/ui/script_queue_commander.py

```diff
@@ -0,0 +1,160 @@
+# This file is part of ts_scriptqueue.
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["ScriptQueueCommander"]
+
+import logging
+
+from lsst.ts.idl.enums.ScriptQueue import Location
+from lsst.ts.idl.enums.Script import ScriptState
+from lsst.ts import salobj
+
+ADD_TIMEOUT = 5  # Timeout for the add command (seconds).
+
+
+class ScriptQueueCommander(salobj.CscCommander):
+    def __init__(self, **kwargs):
+        super().__init__(name="ScriptQueue", **kwargs)
+        self.help_dict[
+            "add"
+        ] = """type path config  # add a script to the end of the queue:
+    • type = s or std for standard, e or ext for external
+    • config = @yaml_file_path or keyword1=value1 keyword2=value2 ..."""
+        self.help_dict["showSchema"] = "type path  # type=s, std, e, or ext"
+        self.script_remote = salobj.Remote(
+            domain=self.domain,
+            name="Script",
+            index=0,
+            readonly=True,
+            include=["logMessage", "state"],
```

Comment:
Nice touch to listen for the `logMessage` and `state` of Scripts... I wonder if there's a way to monitor heartbeats and checkpoints too.... The old UI had a heartbeat monitoring system that would inform if a script stopped publishing HB.